src/parser/jobs/nin/Ninjutsu.js

```diff
@@ -30,8 +39,54 @@ export default class Ninjutsu extends Module {
 		this._rabbitCount++
 	}
 
+	_onDotonCast(event) {
+		this._dotonCasts.push({
+			cast: event,
+			tcj: this.combatants.selected.hasStatus(STATUSES.TEN_CHI_JIN.id), // STDs are only okay under TCJ
+			ticks: [],
+		})
+	}
+
+	_onDotonDamage(event) {
+		// If there are no casts at all, use the damage event to fabricate one
+		if (this._dotonCasts.length === 0) {
+			this._onDotonCast(event)
+		}
+
+		this._dotonCasts[this._dotonCasts.length - 1].ticks.push(event.hits.length) // Track the number of enemies hit per tick
+	}
+
+	_appraiseDotonCasts() {
+		const result = {
+			badTcjs: 0, // TCJ Dotons that fell short of 7 ticks (TCJ Suiton will be more damage)
+			badAoes: 0, // AoE Dotons that fell short of 6 ticks (Katon will be more damage)
+			badStds: 0, // Single-target regular Dotons (do not do this)
+		}
+
+		for (let i = 0; i < this._dotonCasts.length; i++) {
+			const cast = this._dotonCasts[i]
+			if (cast.tcj && cast.ticks.every(tick => tick === 1)) {
+				// If it's a fully single-target TCJ that doesn't land every Doton tick, flag it
+				if (cast.ticks.length < TCJ_DOTON_TICK_TARGET) {
+					result.badTcjs++
+				}
+			} else if (cast.ticks.every(tick => tick > 1)) {
+				// If it's a fully multi-target Doton that misses at least 2 ticks, flag it
+				if (cast.ticks.length < DOTON_TICK_TARGET) {
+					result.badAoes++
+				}
+			} else if (cast.ticks.reduce((accum, value) => accum + value) < JUSTIFIABLE_DOTON_TICKS) {
```

Comment:
I like to be super anal and always provide an initial value to reducers. But that's just me.